src/main/app/index.js

```diff
@@ -231,25 +273,134 @@ class App {
     }
   }
 
-  // TODO(sessions): ...
-  // // Make Mark Text a single instance application.
-  // _makeSingleInstance() {
-  //   if (process.mas) return
-  //
-  //   app.requestSingleInstanceLock()
-  //
-  //   app.on('second-instance', (event, argv, workingDirectory) => {
-  //     // // TODO: Get active/last active window and open process arvg etc
-  //     // if (currentWindow) {
-  //     //   if (currentWindow.isMinimized()) currentWindow.restore()
-  //     //   currentWindow.focus()
-  //     // }
-  //   })
-  // }
+  _openFilesToOpen () {
+    this._openPathList(this._openFilesCache, false)
+  }
+
+  /**
+   * Open the path list in the best window(s).
+   *
+   * @param {string[]} pathsToOpen The path list to open.
+   * @param {boolean} openFilesInSameWindow Open all files in the same window with
+   * the first directory and discard other directories.
+   */
+  _openPathList (pathsToOpen, openFilesInSameWindow=false) {
+    const { _windowManager } = this
+    const { openFilesInNewWindow } = this._accessor.preferences.getAll()
+
+    const fileSet = new Set()
+    const directorySet = new Set()
+    for (const { isDir, path } of pathsToOpen) {
+      if (isDir) {
+        directorySet.add(path)
+      } else {
+        fileSet.add(path)
+      }
+    }
+
+    // Filter out directories that are already opened.
+    for (const window of _windowManager.windows.values()) {
+      if (window.type === WindowType.EDITOR) {
+        const { openedRootDirectory } = window
+        if (directorySet.has(openedRootDirectory)) {
+          window.bringToFront()
+          directorySet.delete(openedRootDirectory)
+        }
+      }
+    }
+
+    const directoriesToOpen = Array.from(directorySet).map(dir => ({ rootDirectory: dir, fileList: [] }))
+    const filesToOpen = Array.from(fileSet)
+
+    // Discard all directories except first one and add files.
```

Comment:
Why discard other directories except the first one?

Reply:
This is the case if new pass `-n|--new-window <dirs> <files>` via command line to a second instance. We want to open only one new window but if more directories are pass we discard these.